packages/timeline-state-resolver/src/integrations/vmix/index.ts

```diff
@@ -1155,8 +1170,10 @@ export class VMixDevice extends DeviceWithState<VMixStateExtended, DeviceOptions
 	): Array<VMixStateCommandWithContext> {
 		let commands: Array<VMixStateCommandWithContext> = []
 
-		commands = commands.concat(this._resolveInputsState(oldVMixState, newVMixState))
+		const inputCommands = this._resolveInputsState(oldVMixState, newVMixState)
+		commands = commands.concat(inputCommands.preTransitionCommands)
 		commands = commands.concat(this._resolveMixState(oldVMixState, newVMixState))
+		commands = commands.concat(inputCommands.postTransitionCommands)
```

Comment:
Shouldn't this appear after resolving overlays state? And shouldn't _isOnAir consider inputs on overlays? What about outputs? You probably wouldn't like an output that is currently fed a list input to show a couple of black frames before it cuts to the next input after the list gets wiped?

Reply:
While this is probably sufficient for your use case, the biggest drawback of this approach is that there's no common definition of _on-air_. An input may appear on an output or in a Mix, but not be considered on-air for numerous reasons. This may cause inconsistencies in preparing inputs just before being taken, and will still require the user to apply some kind of pre-roll.